docs/source/dev/styleguide.rst

```diff
@@ -30,15 +30,19 @@ find include/ share/picongpu/ share/pmacc -iname "*.def" \
 | xargs clang-format-11 -i
 ```
 
-Format Git Staged Changes
-^^^^^^^^^^^^^^^^^^^^^^^^^
+Format Changes With Git
+^^^^^^^^^^^^^^^^^^^^^^^
 
-To format all changed added with `git add` you can run `git clang-format-11`.
+Instead of using the bash command above you can use *Git* together with *clang-format* to format your patched code, only.
+You must extent your local git configuration **once** with all file endings used in PIConGPU:
 
-Format Commits
-^^^^^^^^^^^^^^
+.. code-block:: bash
 
-To format the last commit you can run `git clang-format-11 HEAD~1`.
+   git config --local clangFormat.extensions def,h,cpp,cu,hpp,tpp,kernel,loader,param,unitless
+
+
+To format all changed added with `git add` you can execute `git clang-format-11`.
+The last commit can be formatted by executing `git clang-format-11 HEAD~1`.
```

Comment:
Formatting all changes of the previous commit can be achieved by executing the command `git clang-format-11 HEAD~1`.